### ISSUE TITLE
High load caused by job function::aodIndexUnindexed

### ISSUE THREAD

**Issue by mugge6** [salesagility/SuiteCRM#897]
Detected on SuiteCRM 7.2.2 (but I'm pretty sure it's on current version too)

Problem: 
AODIndex (function::aodIndexUnindexed) job was running forever and causes high cpu load on the server.

Issue:
AOD indexing job was caught in an endless loop for some PDF Documents. After a long time of debugging I've found out that the problem occurs during indexing DocumentRevisions.

Stacktrace:

```
PdfParser->parseTextCommand()
PdfParser->extractTextElements()
PdfParser->extractText()
PdfParser->parseFile()
LuceneUtils->createPDFDocument
AOD_Index->getDocumentForRevision()
AOD_Index->getDocumentForBean()
AOD_Index->index()
```

It looks like that under certain circumstances on PdfParser in method parseTextCommand on line 291 function mb_strpos will return wrong position:
`while (($cur_start_pos = mb_strpos($text, ')', $start_search_end, 'UTF-8')) !== false) {`
After that the parent while-loop will caught in that andles loop, because there is no valid break condition anymore.

My fix:
Extended break condition if returned position is lower or equals than last one, then break out from current loop. This works for me.

PdfParser on line 299:

Code:
`if ($cur_start_pos === false) {`
Fix:
`if ($cur_start_pos === false || $start_search_end >= $cur_start_pos) {`

Possible side effects which still have to be solved:
If my added break condition will match in the middle of the document, then current text element wont be parsed till the end. In that case it's just a quickfix.

Proposed solution to finally fix this issue would be to replace existing PdfParser (it's a very old version from 2013-08-08) with a new version. Which can be found here:
https://github.com/smalot/pdfparser

**Comment by bebbodl** [comment]
You also have to add behind line 275:
old

    while (($cur_start_text = mb_strpos($text, '(', $cur_start_pos, 'UTF-8')) !== false) {

fixed

    while (($cur_start_text = mb_strpos($text, '(', $cur_start_pos, 'UTF-8')) !== false) {
		if ($cur_start_text < $cur_start_pos) break;

**Comment by Mac-Rae** [comment]
This issue has been closed as version 7.8.x of SuiteCRM has reached its end of life and therefore will no longer be receiving updates. We have also been unable to replicate this issue in any newer version of SuiteCRM such as 7.10.x or 7.11.x however if you are able, please leave a comment explaining how and we'll re-open this issue.

For more information about 7.8.x end of life please visit our [blog post](https://suitecrm.com/7-8-x-end-of-life/)
